util/headerreader/blob_client.go

```diff
@@ -91,22 +103,36 @@ func beaconRequest[T interface{}](b *BlobClient, ctx context.Context, beaconPath
 
 	var empty T
 
-	// not really a deep copy, but copies the Path part we care about
-	url := *b.beaconUrl
-	url.Path = path.Join(url.Path, beaconPath)
-
-	req, err := http.NewRequestWithContext(ctx, "GET", url.String(), http.NoBody)
-	if err != nil {
-		return empty, err
-	}
-
-	if b.authorization != "" {
-		req.Header.Set("Authorization", b.authorization)
+	fetchData := func(url url.URL) (*http.Response, error) {
+		url.Path = path.Join(url.Path, beaconPath)
+		req, err := http.NewRequestWithContext(ctx, "GET", url.String(), http.NoBody)
+		if err != nil {
+			return nil, err
+		}
+		if b.authorization != "" {
+			req.Header.Set("Authorization", b.authorization)
+		}
+		resp, err := b.httpClient.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("response returned with status code %d, want 200 (OK)", resp.StatusCode)
+		}
```

Comment:
Can we log some extra details about the request here? Maybe when we merge in https://github.com/OffchainLabs/nitro/pull/2193  we'll have to resolve that.